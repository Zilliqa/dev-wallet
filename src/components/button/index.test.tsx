/**
 * Copyright (c) 2018 - present Zilliqa Research Pte. Ltd.
 *
 * This program is free software: you can redistribute it and/or modify it under the
 * terms of the GNU General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any later
 * version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with
 * this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import { render, cleanup } from '@testing-library/react';
import Button from '.';

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

const levelArray = ['primary', 'secondary', 'tertiary'];
const sizeArray = ['small', 'medium', 'large'];
const disabledArray = [true, false];
const snapshot = (level) => (size) => (disabled) => {
  test('matches the snapshot', () => {
    const onClick = jest.fn();
    const { container } = render(
      <Button
        text={`${level} ${size}`}
        level={level}
        size={size}
        disabled={disabled}
        data-testid={`${level}-${size}`}
        onClick={onClick}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
};

levelArray.forEach((level) =>
  sizeArray.forEach((size) => disabledArray.forEach((disabled) => snapshot(level)(size)(disabled)))
);

test('matches the snapshot', () => {
  const onClick = jest.fn();
  const { container } = render(
    <Button
      text={'before'}
      before={<span>Pre</span>}
      level={'primary'}
      data-testid={'button-before'}
      onClick={onClick}
    />
  );
  expect(container.firstChild).toMatchSnapshot();
});

test('matches the snapshot', () => {
  const onClick = jest.fn();
  const { container } = render(
    <Button
      text={'before'}
      after={<span>Post</span>}
      level={'primary'}
      data-testid={'button-before'}
      onClick={onClick}
    />
  );
  expect(container.firstChild).toMatchSnapshot();
});
