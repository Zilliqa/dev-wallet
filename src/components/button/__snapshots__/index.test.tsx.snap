// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<button
  aria-label="primary small"
  class="sc-bdVaJa iDGSni"
  data-testid="primary-small"
  disabled=""
  type="submit"
>
   
  primary small
   
</button>
`;

exports[`matches the snapshot 2`] = `
<button
  aria-label="primary small"
  class="sc-bdVaJa iDGSni"
  data-testid="primary-small"
  type="submit"
>
   
  primary small
   
</button>
`;

exports[`matches the snapshot 3`] = `
<button
  aria-label="primary medium"
  class="sc-bdVaJa iXJpiU"
  data-testid="primary-medium"
  disabled=""
  type="submit"
>
   
  primary medium
   
</button>
`;

exports[`matches the snapshot 4`] = `
<button
  aria-label="primary medium"
  class="sc-bdVaJa iXJpiU"
  data-testid="primary-medium"
  type="submit"
>
   
  primary medium
   
</button>
`;

exports[`matches the snapshot 5`] = `
<button
  aria-label="primary large"
  class="sc-bdVaJa jTWclE"
  data-testid="primary-large"
  disabled=""
  type="submit"
>
   
  primary large
   
</button>
`;

exports[`matches the snapshot 6`] = `
<button
  aria-label="primary large"
  class="sc-bdVaJa jTWclE"
  data-testid="primary-large"
  type="submit"
>
   
  primary large
   
</button>
`;

exports[`matches the snapshot 7`] = `
<button
  aria-label="secondary small"
  class="sc-bdVaJa jvUehg"
  data-testid="secondary-small"
  disabled=""
  type="submit"
>
   
  secondary small
   
</button>
`;

exports[`matches the snapshot 8`] = `
<button
  aria-label="secondary small"
  class="sc-bdVaJa jvUehg"
  data-testid="secondary-small"
  type="submit"
>
   
  secondary small
   
</button>
`;

exports[`matches the snapshot 9`] = `
<button
  aria-label="secondary medium"
  class="sc-bdVaJa bFmqiI"
  data-testid="secondary-medium"
  disabled=""
  type="submit"
>
   
  secondary medium
   
</button>
`;

exports[`matches the snapshot 10`] = `
<button
  aria-label="secondary medium"
  class="sc-bdVaJa bFmqiI"
  data-testid="secondary-medium"
  type="submit"
>
   
  secondary medium
   
</button>
`;

exports[`matches the snapshot 11`] = `
<button
  aria-label="secondary large"
  class="sc-bdVaJa buxHUR"
  data-testid="secondary-large"
  disabled=""
  type="submit"
>
   
  secondary large
   
</button>
`;

exports[`matches the snapshot 12`] = `
<button
  aria-label="secondary large"
  class="sc-bdVaJa buxHUR"
  data-testid="secondary-large"
  type="submit"
>
   
  secondary large
   
</button>
`;

exports[`matches the snapshot 13`] = `
<button
  aria-label="tertiary small"
  class="sc-bdVaJa gXMpwE"
  data-testid="tertiary-small"
  disabled=""
  type="submit"
>
   
  tertiary small
   
</button>
`;

exports[`matches the snapshot 14`] = `
<button
  aria-label="tertiary small"
  class="sc-bdVaJa gXMpwE"
  data-testid="tertiary-small"
  type="submit"
>
   
  tertiary small
   
</button>
`;

exports[`matches the snapshot 15`] = `
<button
  aria-label="tertiary medium"
  class="sc-bdVaJa jUDwrg"
  data-testid="tertiary-medium"
  disabled=""
  type="submit"
>
   
  tertiary medium
   
</button>
`;

exports[`matches the snapshot 16`] = `
<button
  aria-label="tertiary medium"
  class="sc-bdVaJa jUDwrg"
  data-testid="tertiary-medium"
  type="submit"
>
   
  tertiary medium
   
</button>
`;

exports[`matches the snapshot 17`] = `
<button
  aria-label="tertiary large"
  class="sc-bdVaJa dvElSX"
  data-testid="tertiary-large"
  disabled=""
  type="submit"
>
   
  tertiary large
   
</button>
`;

exports[`matches the snapshot 18`] = `
<button
  aria-label="tertiary large"
  class="sc-bdVaJa dvElSX"
  data-testid="tertiary-large"
  type="submit"
>
   
  tertiary large
   
</button>
`;

exports[`matches the snapshot 19`] = `
<button
  aria-label="before"
  class="sc-bdVaJa iXJpiU"
  data-testid="button-before"
  type="submit"
>
  <span>
    Pre
  </span>
   
  before
   
</button>
`;

exports[`matches the snapshot 20`] = `
<button
  aria-label="before"
  class="sc-bdVaJa iXJpiU"
  data-testid="button-before"
  type="submit"
>
   
  before
   
  <span>
    Post
  </span>
</button>
`;
